convert/convert.go

```diff
@@ -0,0 +1,144 @@
+package convert
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+)
+
+const SegmentSize = 32 * 1024 * 1024
+const BufferSize = 1024 * 1024
+const Workers = 8
```

Comment:
Should this default to GOMAXPROCS?

Reply:
I'm not sure this is a good way. For I/O we want to have multiple request in-flight, regardless of the number of cores. This helps to get better throughput on NVMe devices. For decompression more using the number of cores is optimal, but I'm not sure that using all cores is good approach. Since we have a buffer per core, using more cores uses also more memory for the buffers. Using too much memory typically slow down the copy since you don't fit into L2/L3 cache.

I started with qemu-img approach, defaulting to 8 coroutines and 8 threads in the thread pool, regardless of the number of cores. I tested 1, 2, 4, 8, 12 workers on M2 Max (8 performance cores, 4 efficiency cores) and M1 Pro (8 performance cores, 2 efficiency cores). 8 workers looks good on both machines.

I'll do more testing to evaluate this.

Reply:
Thanks for explanation, could you add that explanation as a code comment too?

Reply:
Sure, good idea.

Reply:
I added more benchmarks results showing that 8 workers gives almost best result in all cases. I think this is a good default value, and users that want to get maximum performance can tweak the options.

For compressed images using number of cores is 20% faster, but I think the way to improve decompression is using faster library like [go-libdeflate](https://github.com/4kills/go-libdeflate). I did a quick test and got 650 MiB/s for single thread instead of 150 MiB/s.